tests/tensorbox/test_functions.py

```diff
@@ -0,0 +1,519 @@
+            fn._get_multi_tensorbox([x, y])
+
+    def test_return_tensorflow_box(self):
+        """Test that TensorFlow is correctly identified as the dispatching library"""
+        x = tf.Variable([1.0, 2.0, 3.0])
+        y = onp.array([0.5, 0.1])
+
+        res = fn._get_multi_tensorbox([y, x])
+        assert res.interface == "tf"
+
+    def test_return_torch_box(self):
+        """Test that Torch is correctly identified as the dispatching library"""
+        x = torch.tensor([1.0, 2.0, 3.0])
+        y = onp.array([0.5, 0.1])
+
+        res = fn._get_multi_tensorbox([y, x])
+        assert res.interface == "torch"
+
+    def test_return_autograd_box(self):
+        """Test that autograd is correctly identified as the dispatching library"""
+        x = np.array([1.0, 2.0, 3.0])
+        y = [0.5, 0.1]
+
+        res = fn._get_multi_tensorbox([y, x])
+        assert res.interface == "autograd"
+
+    def test_return_numpy_box(self):
+        """Test that NumPy is correctly identified as the dispatching library"""
+        x = onp.array([1.0, 2.0, 3.0])
+        y = [0.5, 0.1]
+
+        res = fn._get_multi_tensorbox([y, x])
+        assert res.interface == "numpy"
+
+
+test_data = [
+    (1, 2, 3),
+    [1, 2, 3],
+    onp.array([1, 2, 3]),
+    np.array([1, 2, 3]),
+    torch.tensor([1, 2, 3]),
+    tf.Variable([1, 2, 3]),
+    tf.constant([1, 2, 3]),
+]
+
+
+@pytest.mark.parametrize("t1,t2", list(itertools.combinations(test_data, r=2)))
+def test_allequal(t1, t2):
+    """Test that the allequal function works for a variety
+    of input"""
+    res = fn.allequal(t1, t2)
+
+    if isinstance(t1, tf.Variable):
+        t1 = tf.convert_to_tensor(t1)
+
+    if isinstance(t2, tf.Variable):
+        t2 = tf.convert_to_tensor(t2)
+
+    expected = all(float(x) == float(y) for x, y in zip(t1, t2))
+    assert res == expected
+
+
+@pytest.mark.parametrize("t1,t2", list(itertools.combinations(test_data, r=2)))
+def test_allclose(t1, t2):
+    """Test that the allclose function works for a variety
+    of input"""
+    res = fn.allclose(t1, t2)
+
+    if isinstance(t1, tf.Variable):
+        t1 = tf.convert_to_tensor(t1)
+
+    if isinstance(t2, tf.Variable):
+        t2 = tf.convert_to_tensor(t2)
+
+    expected = all(float(x) == float(y) for x, y in zip(t1, t2))
+    assert res == expected
+
+
+class TestCast:
+    """Tests for the cast function"""
+
+    @pytest.mark.parametrize("t", test_data)
+    def test_cast_numpy(self, t):
+        """Test that specifying a NumPy dtype results in proper casting
+        behaviour"""
+        res = fn.cast(t, onp.float64)
+        assert fn.get_interface(res) == fn.get_interface(t)
+
+        if hasattr(res, "numpy"):
+            # if tensorflow or pytorch, extract view of underlying data
+            res = res.numpy()
+            t = t.numpy()
+
+        assert onp.issubdtype(onp.asarray(t).dtype, onp.integer)

```

Comment:
This tests that the original tensor has an integer dtype. The reason I don't compare to a _specific_ integer dtype (such as `np.int64`) is that the different interfaces have different rules as to which integer dtype they cast to 🙁 But since we are casting to a float, it is sufficient to simply check that the input was not a float (it's an integer)